codegen/gateway.go

```diff
@@ -115,12 +117,38 @@ type ClientClassConfig struct {
 	Dependencies Dependencies           `json:"dependencies"`
 }
 
+// helper struct to pull out the fixture config
+type clientClassConfigFixture struct {
+	Config struct {
+		Fixture *Fixture `json:"fixture"`
+	} `json:"config"`
+}
+
 // Dependencies lists all dependencies of a module
 type Dependencies struct {
 	Client []string `json:"client"`
 	//	Service []string `json:"service"`  // example extension
 }
 
+// Fixture specifies client fixture import path and all scenarios
+type Fixture struct {
+	ImportPath string              `json:"importPath"`
+	Scenarios  map[string][]string `json:"scenarios"`
```

Comment:
`// Scenarios is a map from zanzibar's exposed method name to a list of user-defined fixture scenarios for a client.`

Reply:
done